mixer/adapter/cloudwatch/logHandler_test.go

```diff
@@ -0,0 +1,337 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY Type, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudwatch
+
+import (
+	"errors"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs/cloudwatchlogsiface"
+
+	"istio.io/istio/mixer/adapter/cloudwatch/config"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/pkg/adapter/test"
+	"istio.io/istio/mixer/template/logentry"
+)
+
+type mockCloudWatchLogsClient struct {
+	cloudwatchlogsiface.CloudWatchLogsAPI
+	resp          *cloudwatchlogs.DescribeLogStreamsOutput
+	describeError error
+}
+
+func (m *mockCloudWatchLogsClient) PutLogEvents(input *cloudwatchlogs.PutLogEventsInput) (*cloudwatchlogs.PutLogEventsOutput, error) {
+	return &cloudwatchlogs.PutLogEventsOutput{}, nil
+}
+
+func (m *mockCloudWatchLogsClient) DescribeLogStreams(input *cloudwatchlogs.DescribeLogStreamsInput) (*cloudwatchlogs.DescribeLogStreamsOutput, error) {
+	return m.resp, m.describeError
+}
+
+type mockFailCloudWatchLogsClient struct {
```

Comment:
super nit: just calling this `failLogsClient` would be sufficient.

Reply:
okay